lib/3scale_toolbox/commands/copy_command/copy_service.rb

```diff
@@ -4,83 +4,70 @@
 module ThreeScaleToolbox
   module Commands
     module CopyCommand
-      module CopyServiceSubcommand
-        extend ThreeScaleToolbox::Command
+      class CopyServiceSubcommand < Cri::CommandRunner
+        include ThreeScaleToolbox::Command
         def self.command
           Cri::Command.define do
             name        'service'
             usage       'service [opts] -s <src> -d <dst> <service_id>'
             summary     'Copy service'
             description 'Will create a new services, copy existing proxy settings, metrics, methods, application plans and mapping rules.'
 
-            required  :s, :source, '3scale source instance. Format: "http[s]://<provider_key>@3scale_url"'
-            required  :d, :destination, '3scale target instance. Format: "http[s]://<provider_key>@3scale_url"'
-            required  :t, 'target_system_name', 'Target system name'
+            option  :s, :source, '3scale source instance. Format: "http[s]://<provider_key>@3scale_url"', argument: :required
+            option  :d, :destination, '3scale target instance. Format: "http[s]://<provider_key>@3scale_url"', argument: :required
+            option  :t, 'target_system_name', 'Target system name', argument: :required
+            param   :service_id
 
-            run do |opts, args, _|
-              CopyServiceSubcommand.run opts, args
-            end
+            runner CopyServiceSubcommand
           end
         end
 
-        def self.run(opts, args)
-          source      = fetch_required_option(opts, :source)
-          destination = fetch_required_option(opts, :destination)
-          system_name = fetch_required_option(opts, :target_system_name)
-          insecure    = opts[:insecure] || false
-          exit_with_message 'error: missing service_id argument' if args.empty?
-          service_id = args[0]
-          copy_service(service_id, source, destination, system_name, insecure)
-        end
-
-        def self.exit_with_message(message)
-          puts message
-          exit 1
-        end
-
-        def self.fetch_required_option(options, key)
-          options.fetch(key) { exit_with_message "error: Missing argument #{key}" }
+        def run
+          source      = fetch_required_option(:source)
+          destination = fetch_required_option(:destination)
+          system_name = fetch_required_option(:target_system_name)
+          copy_service(arguments[:service_id], source, destination, system_name)
```

Comment:
This is really a nitpick, but I think a whitespace before this line would be really useful.